### ISSUE TITLE
pulumi stack select call not working for org stacks

### ISSUE THREAD

**Issue by devcauldron** [pulumi/actions#777]
### What happened?

Using `pulumi/actions@v3` on a manually run GH action, the call to pulumi select keeps giving an error stating that the stack cannot be found (printed error is missing the 2nd part of the stack name).

All of my GH Actions now cannot get past this step and so nothing is working.

Here is the output of the workflow step:

```sh
code: -2
 stdout: 
 stderr: Command failed with exit code 255: pulumi stack select --stack kettleos/base/dev --non-interactive
error: no stack named 'kettleos/dev' found
 err?: Error: Command failed with exit code 255: pulumi stack select --stack kettleos/base/dev --non-interactive
error: no stack named 'kettleos/dev' found
```

As you can see the `--stack` argument has all 3 parts of the stack name (`kettleos/base/dev`) in it but the error message prints out and it is missing the 2nd part (`'kettleos/dev'`)

### Steps to reproduce

Here is an example Github Workflow file modified from one of ours in our infrastructure repo that has several projects each run as a separate Pulumi Stack

```yaml
name: Pulumi Refresh (on demand)
on:
  workflow_dispatch:
    inputs:
      project:
        type: choice
        options:
          - primary
          - base
          - keycloak
        default: base
        description: 'Which Pulumi Project (Stack)'
        required: true
      environment:
        type: choice
        options:
          - dev
          - prod
        description: 'Select which environment for the Stack to Refresh'
        required: true
        default: dev
      preview:
        type: boolean
        description: 'Display a Preview of what to do or Perform the Refresh'
        default: true
        required: true

jobs:
  refresh:
    name: Refresh
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: '16.17.0'
      - name: Configure AWS Credentials
        uses: aws-actions/configure-aws-credentials@v1-node16
        with:
          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
          aws-region: "us-east-1"
          role-to-assume: CICD
          role-duration-seconds: 1800
          role-session-name: github-pulumi-refresh
      - run: npm install
        working-directory: ${{ inputs.project }}
      - uses: pulumi/actions@v3
        if: ${{ inputs.preview }}
        with:
          command: preview
          stack-name: kettleos/${{ inputs.project }}/${{ inputs.environment }}
          work-dir: ${{ inputs.project }}
          pulumi-version: "3.46.1"
        env:
          PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
      - uses: pulumi/actions@v3
        if: ${{ !inputs.preview }}
        with:
          command: refresh
          stack-name: kettleos/${{ inputs.project }}/${{ inputs.environment }}
          work-dir: ${{ inputs.project }}
          pulumi-version: "3.46.1"
        env:
          PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
```

### Expected Behavior

Workflow will run, select the correct stack (this used to work but has stopped) and execute whichever Pulumi command in the workflow against that stack.

### Actual Behavior

Pulumi select is getting an error and that says it cannot find a stack with the provided name and blocks the rest of the workflow from completing.

### Output of `pulumi about`

Repo is a monorepo with several stacks in subfolders.

Here is the output at the root:
```sh
CLI
Version      3.46.1
Go Version   go1.19.2
Go Compiler  gc

Host
OS       darwin
Version  12.5.1
Arch     x86_64

Backend
Name           pulumi.com
URL            https://app.pulumi.com/devcauldron
User           devcauldron
Organizations  devcauldron, kettleos

Pulumi locates its logs in /var/folders/lq/wqz1b0jn4r72vql_dk3l213c0000gn/T/ by default
warning: Failed to read project: no Pulumi.yaml project file found (searching upwards from /Users/jeff/dev/kettlespace-eng/kettleos-infra). If you have not created a project yet, use `pulumi new` to do so: no project file found
warning: Failed to get information about the current stack: no Pulumi.yaml project file found (searching upwards from /Users/jeff/dev/kettlespace-eng/kettleos-infra). If you have not created a project yet, use `pulumi new` to do so: no project file found
```

Here is the output of one of the subfolders in question (`base`):
```sh
CLI
Version      3.46.1
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.20.0
aws         5.10.0
docker      3.5.0
eks         0.42.7
kubernetes  3.22.1
kubernetes  3.20.2
nodejs      unknown
vault       3.6.0

Host
OS       darwin
Version  12.5.1
Arch     x86_64

This project is written in nodejs: executable='/Users/jeff/.asdf/shims/node' version='v16.17.0'

Backend
Name           pulumi.com
URL            https://app.pulumi.com/devcauldron
User           devcauldron
Organizations  devcauldron, kettleos

Dependencies:
NAME                VERSION
@pulumi/awsx        0.40.1
@pulumi/eks         0.42.7
@pulumi/kubernetes  3.22.1
@pulumi/pulumi      3.46.1
@pulumi/vault       3.6.0
yaml                2.1.3
@pulumi/aws         5.20.0

Pulumi locates its logs in /var/folders/lq/wqz1b0jn4r72vql_dk3l213c0000gn/T/ by default
warning: Failed to get information about the current stack: performing HTTP request: Get "https://api.pulumi.com/api/stacks/kettleos/base/dev": net/http: TLS handshake timeout
```

### Additional context

I was able to run `pulumi refresh` from local CLI in order to refresh my stacks.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by RobbieMcKinstry** [comment]
@devcauldron is this new behavior, or have you always had this problem? If you pin your version to v3.19.1, do you still have this error?

I wonder if this is from recent automation API changes, or if we're just chopping up the stack name incorrectly.

**Comment by devcauldron** [comment]
Thanks @RobbieMcKinstry I will try v`3.19.1` and report back

**Comment by RobbieMcKinstry** [comment]
Thank you!

**Comment by devcauldron** [comment]
@RobbieMcKinstry are you sure `3.19.1` is the version I should try? When running it says not found

I also changed the `package.json` and `package-lock.json` in the working directory as well as tried removing the npm install step. I get this same output every time:
```sh
Run pulumi/actions@v3
  with:
    command: preview
    stack-name: kettleos/base/dev
    work-dir: base
    pulumi-version: 3.19.1
    comment-on-pr: false
    comment-on-pr-number: 0
    github-token: ***
    parallel: [2](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472021644/jobs/5802283008#step:8:2)14748[3](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472021644/jobs/5802283008#step:8:3)6[4](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472021644/jobs/5802283008#step:8:4)[7](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472021644/jobs/5802283008#step:8:7)
    target-dependents: false
    refresh: false
    upsert: false
    edit-pr-comment: true
    color: auto
  env:
    AWS_ACCESS_KEY_ID_NAME: DEV_AWS_ACCESS_KEY_ID
    AWS_SECRET_ACCESS_KEY_NAME: DEV_AWS_SECRET_ACCESS_KEY
    GRAFANA_ADMIN_USER_NAME: DEV_GRAFANA_ADMIN_USER
    GRAFANA_ADMIN_PASSWORD_NAME: DEV_GRAFANA_ADMIN_PASSWORD
    GRAFANA_CLIENT_SECRET_NAME: DEV_GRAFANA_CLIENT_SECRET
    AWS_DEFAULT_REGION: us-east-1
    AWS_REGION: us-east-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
    AWS_SESSION_TOKEN: ***
    PULUMI_ACCESS_TOKEN: ***
  
Configured range: 3.1[9](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472021644/jobs/5802283008#step:8:9).1
Error: Could not find a version that ***tisfied the version range

**Comment by RobbieMcKinstry** [comment]
Hi @devcauldron. 
Sorry for the miscommunication. I was suggesting pinning the version of the Pulumi GitHub Action, not the Pulumi SDK. You shouldn't be changing values in your package.json, nor the field `pulumi-version`.

In the code you gave me:

```
- name: Run Pulumi
   uses: pulumi/actions@v3.19.1
   with:
    command: preview
    stack-name: kettleos/base/dev
    work-dir: base
    comment-on-pr: false
```
Notice how I removed the field `pulumi-version`, and adjusted the version of the Action in the `uses:` statement.

More broadly, however, did your GitHub Action start failing recently by itself? Did it work previously?
I'm trying to ascertain if this bug is a result of a change we released in the GitHub Action recently, or if it predates our most recent release.

**Comment by devcauldron** [comment]
Thanks again @RobbieMcKinstry - apologies for not understanding. I will try the version on the action rather than the SDK (which makes more sense).

> 
> More broadly, however, did your GitHub Action start failing recently by itself? Did it work previously? I'm trying to ascertain if this bug is a result of a change we released in the GitHub Action recently, or if it predates our most recent release.

In answer to your question above, it did work previously but hadn't been run in some time so I can't say exactly when it failed. This is for infrastructure that isn't changing much as opposed to our apps that do and are deployed more frequently but they aren't using the pulumi action.

**Comment by devcauldron** [comment]
@RobbieMcKinstry it failed again for the same reason. Here is the output:

```sh
Run pulumi/actions@v3.19.1
  with:
    command: preview
    stack-name: kettleos/base/dev
    work-dir: base
    pulumi-version: 3.46.1
    comment-on-pr: false
    github-token: ***
    parallel: [2](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472943249/jobs/5804398597#step:8:2)14748[3](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472943249/jobs/5804398597#step:8:3)6[4](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472943249/jobs/5804398597#step:8:4)7
    target-dependents: false
    refresh: false
    upsert: false
    edit-pr-comment: true
    color: auto
  env:
    AWS_ACCESS_KEY_ID_NAME: DEV_AWS_ACCESS_KEY_ID
    AWS_SECRET_ACCESS_KEY_NAME: DEV_AWS_SECRET_ACCESS_KEY
    GRAFANA_ADMIN_USER_NAME: DEV_GRAFANA_ADMIN_USER
    GRAFANA_ADMIN_PASSWORD_NAME: DEV_GRAFANA_ADMIN_PASSWORD
    GRAFANA_CLIENT_SECRET_NAME: DEV_GRAFANA_CLIENT_SECRET
    AWS_DEFAULT_REGION: us-east-1
    AWS_REGION: us-east-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
    AWS_SESSION_TOKEN: ***
    PULUMI_ACCESS_TOKEN: ***
  
Configured range: 3.46.1
Matched version: v3.46.1
Install destination is /home/runner/.pulumi
Successfully deleted pre-existing /home/runner/.pulumi/bin
/usr/bin/tar xz --warning=no-unknown-keyword --overwrite -C /home/runner/.pulumi -f /home/runner/work/_temp/07dad7ff-e884-4cc[5](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472943249/jobs/5804398597#step:8:5)-8300-838a8b2b7e3[6](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472943249/jobs/5804398597#step:8:6)
Error: code: -2
 stdout: 
 stderr: Command failed with exit code [25](https://github.com/kettlespace-eng/kettleos-infra/actions/runs/3472943249/jobs/5804398597#step:8:25)5: pulumi stack select kettleos/base/dev --non-interactive
error: no stack named 'kettleos/dev' found
 err?: Error: Command failed with exit code 255: pulumi stack select kettleos/base/dev --non-interactive
error: no stack named 'kettleos/dev' found
```

**Comment by devcauldron** [comment]
@RobbieMcKinstry need some help here. We pay for Pulumi as a service and we're not able to use it now. We need to run some updates for fixes on this infrastructure.

Should I just be ticking back every version number until this works?
Should I stop using this action and run from scripts in a shell as an Action instead?

**Comment by RobbieMcKinstry** [comment]
Hi @devcauldron , sorry again that you're having trouble with this. For time-sensitive issues, I suggest reaching out to support staff if your account has enterprise support. If you don't, the [Pulumi Community Slack](https://slack.pulumi.com/) tends to produce faster answers than issues.

As an immediate workaround, you can always access Pulumi in CI by installing it with the [setup-pulumi](https://github.com/pulumi/setup-pulumi) action and running the desired command directly.

Here's an example

```yaml
steps:
- name: "Install Pulumi CLI"
  uses: pulumi/setup-pulumi@v2

- name: "Run Pulumi CLI"
   run: pulumi preview # Use whatever Pulumi command you want here.
  env:
    AWS_ACCESS_KEY_ID_NAME: DEV_AWS_ACCESS_KEY_ID
    AWS_SECRET_ACCESS_KEY_NAME: DEV_AWS_SECRET_ACCESS_KEY
    GRAFANA_ADMIN_USER_NAME: DEV_GRAFANA_ADMIN_USER
    GRAFANA_ADMIN_PASSWORD_NAME: DEV_GRAFANA_ADMIN_PASSWORD
    GRAFANA_CLIENT_SECRET_NAME: DEV_GRAFANA_CLIENT_SECRET
    AWS_DEFAULT_REGION: us-east-1
    AWS_REGION: us-east-1
    AWS_ACCESS_KEY_ID: ***
    AWS_SECRET_ACCESS_KEY: ***
    AWS_SESSION_TOKEN: ***
    PULUMI_ACCESS_TOKEN: ***
```

--------------------------------

Are you able to run this project from the CLI?

It's not obvious to me where the bug is coming from. From tracing the code, it appears the fully qualified stack name is correctly sent to the Automation API. I'd want to confirm that with a reproduction on my device or with a unit test. If that's the case, then the bug must be in the Node Automation API, which sounds unlikely to me since Automation API is widely used, and others aren't reporting this bug, so I suspect there's an issue in the Action that I'm not seeing. I would need to look into it more closely.

-------------------------------

> Should I just be ticking back every version number until this works?

It might be gainful to both of us if you could identify the version that introduced this issue for you. With the information we have now, we know the version was somewhere between v3.0 and v3.19.1, which means the bug was introduced between Apr 20, 2021 and September 6, 2022. That's a long time to search. If we knew the version introducing the bug, we'd narrow the search space considerably. But I wouldn't recommend ticking back every version number. Binary search would be more efficient. Still, I think the approach I mentioned above would be just as fruitful -- we would just need additional time to dig into the code.

In the future, I recommend pinning the full version of this GitHub Action for consistent behavior. If using `pulumi/actions@v3` and pinning the major version only, GitHub Actions will pick the newest possible release, which could potentially introduce buggy behavior. We adhere to Semver guarantees for this Action, but bugs do happen.

Here's another sample of how to pin to the full version.

```typescript
- name: Run Pulumi
   uses: pulumi/actions@v3.19.1
   with:
    command: preview
    stack-name: kettleos/base/dev
    work-dir: base
    comment-on-pr: false
```

**Comment by devcauldron** [comment]
@RobbieMcKinstry updating this issue prior to posting in Slack in an effort to get help on this.

I tried using the action versions in this order, all with the exact same results:

1. `3.17.0`
2. `3.12.0`
3. `3.9.1`
4. `3.6.0`

Additionally, after reviewing all of our actions, I found the last time it worked was Oct 26, 2022, we have automated actions that run `pulumi preview` on push to branch, and all 3 ran successfully (3 different projects within same monorepo). To be clear, the first failure occurred on Nov 14, 2022.

After recognizing that above, I went back to action v `3.19.1` b/c it allowed choosing the `pulumi-version` and tried the version that was successful in those jobs, `3.44.1` which now fails as well.

In each case, the `stack-name` is showing the expected fully qualified stack name in the printout of the action run but (when available, earlier versions of pulumi/action didn't log to output) the command errors with the project (middle term) missing:
```sh
Logging into the Pulumi Service backend.
Error: code: -2
 stdout: 
 stderr: Command failed with exit code 255: pulumi stack select --stack kettleos/kettlespace/dev --non-interactive
error: no stack named 'kettleos/dev' found
 err?: Error: Command failed with exit code 255: pulumi stack select --stack kettleos/kettlespace/dev --non-interactive
error: no stack named 'kettleos/dev' found
```
